components/shapediver/parameter/select/SelectCarouselComponent.tsx

```diff
@@ -0,0 +1,175 @@
+import {Carousel} from "@mantine/carousel";
+import "@mantine/carousel/styles.css";
+import {
+	Box,
+	Card,
+	Image,
+	MantineThemeComponent,
+	MantineThemeOverride,
+	StyleProp,
+	Text,
+	useProps,
+} from "@mantine/core";
+import React, {useCallback, useMemo} from "react";
+import ThemeProvider from "../../ui/ThemeProvider";
+import classes from "./SelectCarouselComponent.module.css";
+import {SelectComponentProps} from "./SelectComponent";
+
+interface StyleProps {
+	slideSize: StyleProp<string | number>;
+	slideGap: StyleProp<string | number>;
+	height: string | number;
+	themeOverride?: MantineThemeOverride;
+}
+
+export const defaultStyleProps: Partial<StyleProps> = {
+	slideSize: {base: "100%", "200px": "50%", "500px": "33.333333%"},
+	slideGap: {base: 0, "200px": "md"},
+	height: "auto",
+};
+
+type SelectCarouselComponentThemePropsType = Partial<StyleProps>;
+
+export function SelectCarouselComponentThemeProps(
+	props: SelectCarouselComponentThemePropsType,
+): MantineThemeComponent {
+	return {
+		defaultProps: props,
+	};
+}
+
+/**
+ * Functional component that displays selectable items in a carousel with images.
+ * Images can be displayed with or without title/description.
+ * The carousel is responsive based on container width.
+ */
+export default function SelectCarouselComponent(
+	props: SelectComponentProps & SelectCarouselComponentThemePropsType,
+) {
+	// style properties
+	const {
+		value,
+		onChange,
+		items,
+		disabled,
+		itemData,
+		settings,
+		slideSize,
+		slideGap,
+		height,
+		themeOverride,
+	} = useProps("SelectCarouselComponent", defaultStyleProps, props);
+
+	const imageFit = settings?.imageFit || "cover";
+	const withIndicators = settings?.withIndicators !== false;
+
+	// Transform items array into the format expected by the component
+	const carouselItems = useMemo(
+		() =>
+			items.map((item) => {
+				const data = itemData?.[item];
+
+				return {
+					value: item,
+					label: data?.displayname || item,
+					description: data?.description,
+					imageUrl: data?.imageUrl,
+					color: data?.color,
+				};
+			}),
+		[items, itemData],
+	);
+
+	// Handle card selection
+	const handleCardClick = useCallback(
+		(itemValue: string, disabled: boolean | undefined) => {
+			if (!disabled) {
+				onChange(itemValue);
+			}
+		},
+		[onChange],
+	);
+
+	const getCardStyle = useCallback((item: (typeof carouselItems)[0]) => {
+		return {
+			"--card-outline-color":
+				item.color || "var(--mantine-primary-color-filled)",
+		};
+	}, []);
+
+	const hasText = useCallback((item: (typeof carouselItems)[0]) => {
+		return item.label || item.description;
+	}, []);
+
+	const getText = useCallback((item: (typeof carouselItems)[0]) => {
+		const text = (
+			<div>
+				{item.label && <Text size="md">{item.label}</Text>}
+				{item.description && (
+					<Text c="dimmed" size="sm">
+						{item.description}
+					</Text>
+				)}
+			</div>
+		);
+
+		return themeOverride ? (
```

Comment:
@snabela the themeOverride may be excessive here for font sizes.

Reply:
@LemToUp agreed, we can remove the `themeOverride` property again from this component